### PR TITLE
I2c encoder support

### PR DESCRIPTION
**Pull request by giovanniborella**
This PR adds support for the Adafruit I2C QT Rotary Encoder with NeoPixel break-out board over i2c. 

Only basic functionality like the stock encoder extension:
* invertable
* on_move, on_press keypress.

Still needs work for implementing NeoPixel-capabilities

### CODE REVIEW

**Comment by kdb424** [comment]
Try running `make fix-formatting fix-isort test` and see if that returns a clean test. We should be able to review once that test passes.

**Comment by kdb424** [comment]
isort is still not pleased, and looks like there are some other changes needed. We can get to those after you intend to come out of draft state, but the line I showed before should sort that, and has been tested on Mac and Linux. I'm not sure if anyone has tested Windows.

**Comment by giovanniborella** [comment]
@kdb424 I'm trying to get make up and running on WSL, will pull out of draft when test is successfull

**Comment by kdb424** [comment]
Looks generally great. Don't mind us nitpicking the small details here and there. Thanks so much for adding support for this, and we'll get it merged as soon as we can!

**Comment by Tonasz** [comment]
A lot of code is duplicated with `encoder.py` file. Most of code could be reused. For example there could be `AbstractEncoder` and inheriting from him `GPIOEncoder` (I don't know what's the right name) and `I2CEncoder` and they would differ only where they need. What do you think about it?
Also, I see that doc example use `from kmk.modules.i2c_encoder import i2cEncoderHandler` but that's `EncoderHandler` in `i2c_encoder.py` file.

**Comment by giovanniborella** [comment]
Good point @Tonasz! - ill get on merging the two

**Comment by giovanniborella** [comment]
I dont have a GPIO encoder setup, so i'm afraid that part is untested

File: kmk/modules/encoder.py
Comment (Tonasz) [inline kmk/modules/encoder.py:94]:
```suggestion
        new_button_state = self.pin_button.get_value()
        if new_button_state != self._button_state:
            self._button_state = new_button_state
            if self.on_button_do is not None:
                self.on_button_do(self.get_state())
```

**Review by Tonasz — CHANGES_REQUESTED** [review]
It looks a lot cleaner now :)
Despite reintroducing 2 small, but already fixed in the past issues, **it looks good to me**.
I don't have I2C encoders to test it. I got one classic encoder on my keyboard and it works fine. Somebody with multiple encoders could confirm it working, but if mentioned issue is fixed, it should be fine.

**Comment by giovanniborella** [comment]
I merged @Tonasz' suggestions

**Comment by kdb424** [comment]
I'll leave the final call to Tonasz before I check for any final merge blockers.

**Review by Tonasz — APPROVED** [review]
I'm not able right now to test how it will behave with multiple encoders (but it should be alright after latest changes) but this looks good to me :) I don't know other I2C encoders and how compatible they are with each other, so if there's gonna be a need, we will have to do some additional changes.

**Review by Tonasz — APPROVED** [review]
LGTM :) (see above comment)

**Review by kdb424 — APPROVED** [review]

**Pull request merged**
